---
title: "SLE777_Renee Carroll_Assessment 4"
author: "Renee Carroll"
date: "2025-09-25"
output: 
  html_document:
    toc: yes
    fig_width: 7
    fig_height: 7
theme: yeti
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

## Part 1
The file “gene_expression.tsv” contains RNA-seq count data for three samples of interest.

# 1. Read in the file, making the gene identifiers the row names. Show a table of values for the first six genes.

```{r download files from GitHub, and show first 6 genes}
URL1 = "https://raw.githubusercontent.com/ghazkha/Assessment4/refs/heads/main/gene_expression.tsv"
download.file(URL1, destfile="gene_expression.tsv")
gene_expression <- read.delim("gene_expression.tsv", header = TRUE, row.names = 1, sep = "\t", stringsAsFactors = FALSE)
head(gene_expression)
```

# 2. Make a new column which is the mean of the other columns. Show a table of values for the first six genes.

```{r mean saved as new column}
gene_expression$mean_expression <- rowMeans(gene_expression)
head(gene_expression)
```

# 3. List the 10 genes with the highest mean expression

```{r 10 genes with highest mean - whole table}
ordered_expression <- gene_expression[order(-gene_expression$mean_expression),]
genes_highest_expression <- head(ordered_expression, 10)
genes_highest_expression
```

# Listing just the 10 gene names

```{r 10 genes with highest mean - just names as a list}
rownames(genes_highest_expression)
```

# 4. Determine the number of genes with a mean < 10

```{r number of genes with low expression < 10}
low_exp_genes <- subset(gene_expression, mean_expression<10)
nrow(low_exp_genes)
```

# 5. Make a histogram plot of the mean values and include it into your report.

```{r histogram}
hist(gene_expression$mean_expression, breaks=1000, main = "Mean Gene Expression", xlab="Mean Gene Expression", col="dodgerblue2")
```

As the data is highly positively skewed, it is best represented on a logarithmic scale.


```{r histogram of log of values}
gene_subset_expressionhigherthan10 <- subset(gene_expression, mean_expression>10)
log_gene_subset_expressionhigherthan10 <- log(gene_subset_expressionhigherthan10)
hist(log_gene_subset_expressionhigherthan10$mean_expression, breaks=50, main = "Mean Gene Expression", xlab = "Log Mean Gene Expression", col = "dodgerblue2")
```

Try that again, using log+1, as log of 0 gives infinity... does this make a difference to how the graph looks?


```{r histogram of log of values+1}
gene_subset_expressionhigherthan10 <- subset(gene_expression, mean_expression>10)
log_gene_subset_expressionhigherthan10 <- log(gene_subset_expressionhigherthan10+1)
hist(log_gene_subset_expressionhigherthan10$mean_expression, breaks=50, main = "Mean Gene Expression", xlab = "Log Mean Gene Expression", col = "dodgerblue2")
```


The file “growth_data.csv” contains measurements for tree circumference growing at two sites, control site and treatment site which were planted 20 years ago.

# 6. Import this csv file into an R object. What are the column names?

```{r Import csv file}
URL2 = "https://raw.githubusercontent.com/ghazkha/Assessment4/refs/heads/main/growth_data.csv"
download.file(URL2, destfile="growth_data.csv")
growth_data <- read.csv("growth_data.csv") # read.csv automatically treats the first row as headings, therefore you don't need to specify it in the code
```

```{r determine column names}
colnames(growth_data)
```





# 7. Calculate the mean and standard deviation of tree circumference at the start and end of the study at both sites.

```{r northeast site, mean of tree circumferences, start of study}
mean(subset(growth_data, Site == "northeast")$Circumf_2005_cm)
```

```{r northeast site, standard deviation of tree circumferences, start of study}
sd(subset(growth_data, Site == "northeast")$Circumf_2005_cm)
```

```{r northeast site, mean of tree circumferences, end of study}
mean(subset(growth_data, Site == "northeast")$Circumf_2020_cm)
```

```{r northeast site, standard deviation of tree circumferences, end of study}
sd(subset(growth_data, Site == "northeast")$Circumf_2020_cm)
```

```{r southwest site, mean of tree circumferences, start of study}
mean(subset(growth_data, Site == "southwest")$Circumf_2005_cm)
```

```{r southwest site, standard deviation of tree circumferences, start of study}
sd(subset(growth_data, Site == "southwest")$Circumf_2005_cm)
```

```{r southwest site, mean of tree circumferences, end of study}
mean(subset(growth_data, Site == "southwest")$Circumf_2020_cm)
```

```{r southwest site, standard deviation of tree circumferences, end of study}
sd(subset(growth_data, Site == "southwest")$Circumf_2020_cm)
```

# 8. Make a box plot of tree circumference at the start and end of the study at both sites.

```{r boxplot of tree circumference at the start and end of the study at both sites}
boxplot(subset(growth_data, Site == "northeast")$Circumf_2005_cm, subset(growth_data, Site == "northeast")$Circumf_2020_cm, subset(growth_data, Site == "southwest")$Circumf_2005_cm, subset(growth_data, Site == "southwest")$Circumf_2020_cm, names = c("Northeast 2005", "Northeast 2020", "Southwest 2005", "Southwest 2020"), ylab = "Circumference (cm)", main = "Increase in tree girth between 2005 and 2020, at each study site")
```

# 9. Calculate the mean growth over the last 10 years at each site.

```{r mean growth over the last 10 years, northeast site}
growth_data$growth_2010_2020 <- growth_data$Circumf_2020_cm-growth_data$Circumf_2010_cm
mean(subset(growth_data, Site == "northeast")$growth_2010_2020)
```
```{r mean growth over the last 10 years, southwest site}
mean(subset(growth_data, Site == "southwest")$growth_2010_2020)
```

# 10. Use the t.test to estimate the p-value that the 10 year growth is different at the two sites.
The t.test() function returns an object containing various information, including:
t-statistic: The calculated t-value.
df: Degrees of freedom.
p-value: The probability of observing a t-statistic as extreme as, or more extreme than, the calculated one, assuming the null hypothesis is true. A small p-value (typically < 0.05) suggests evidence against the null hypothesis.
confidence interval: A range within which the true population mean difference (for two-sample tests) or population mean (for one-sample tests) is likely to fall.
sample means: The means of the groups or sample.

Assumptions:
T-tests rely on assumptions like normality of data (or residuals for two-sample tests) and, for independent samples, potentially equality of variances. You may need to check these assumptions before interpreting the results.

```{r test for normality in the northeast data}
hist(subset(growth_data, Site == "northeast")$growth_2010_2020, breaks = 20)
```

```{r test for normality in the southwest data}
hist(subset(growth_data, Site == "southwest")$growth_2010_2020, breaks = 20)
```


```{r t.test, not assuming equal variance (Welch t.test)}
t.test(growth_2010_2020 ~ Site, data = growth_data, var.equal = FALSE) #var.equal = TRUE: Include this if you assume equal variances between the groups (for a Student's t-test). Omit or set to FALSE for Welch's t-test (which doesn't assume equal variances).
```
```{r t.test, assuming equal variance}
t.test(growth_2010_2020 ~ Site, data = growth_data, var.equal = TRUE) #var.equal = TRUE: Include this if you assume equal variances between the groups (for a Student's t-test). Omit or set to FALSE for Welch's t-test (which doesn't assume equal variances).
```

## Part 2 Examining biological sequence diversity

In order to analyse sequence data, there are two packages we need to first install and load.

Seqinr is a package designed to process and analyse sequence data.
R.utils is a package containing general utilities like zip and unzip

```{r installing packages}
#install.packages("seqinr")
#install.packages("R.utils")
```


```{r loading packages}
suppressPackageStartupMessages({
  library("seqinr")
  library("R.utils")
})
```

# 1. Download the whole set of coding DNA sequences for E. coli and your organism of interest. How many coding sequences are present in these organisms? Present this in the form of a table. Describe any differences between the two organisms. 

```{r Download E.coli genome}
URL3 = "https://ftp.ensemblgenomes.ebi.ac.uk/pub/bacteria/release-62/fasta/bacteria_0_collection/escherichia_coli_str_k_12_substr_mg1655_gca_000005845/cds/Escherichia_coli_str_k_12_substr_mg1655_gca_000005845.ASM584v2.cds.all.fa.gz"
download.file(URL3, destfile="ecoli_cds.fa.gz")
gunzip("ecoli_cds.fa.gz")
```

```{r Download organism of interest, Acetobacter aceti GCA_002005445}
URL4 = "https://ftp.ensemblgenomes.ebi.ac.uk/pub/bacteria/release-62/fasta/bacteria_80_collection/acetobacter_aceti_gca_002005445/cds/Acetobacter_aceti_gca_002005445.ASM200544v1.cds.all.fa.gz"
download.file(URL4, destfile="aaceti_cds.fa.gz")
gunzip("aaceti_cds.fa.gz")
```
```{r reading in the two sequence data files}
ecoli_cds <- seqinr::read.fasta("ecoli_cds.fa")
aaceti_cds <- seqinr::read.fasta("aaceti_cds.fa")
```

```{r}
str(head(ecoli_cds))
```

```{r}
str(head(aaceti_cds))
```



```{r Table of the number of coding sequences present in these organisms}
install.packages("data.table")
library(data.table)
cds_table_number <- data.table(Organism = c("E.coli", "A.aceti"), "Number of Coding Sequences" = c(length(ecoli_cds), length(aaceti_cds)))
print(cds_table_number)
```
# Describe any differences between the two organisms. 
```{r percentage increase in number of coding sequences}
percentage_increase <- (length(ecoli_cds)-length(aaceti_cds))/length(aaceti_cds)*100
print(percentage_increase)
```
The organism E.coli contains 24.6% more genes than the organism A.aceti, suggesting E.coli is a more complex organism.

# 2. How much coding DNA is there in total for these two organisms? Present this in the form of a table. Describe any differences between the two organisms.
```{r Table of the total amount of coding DNA for the two organisms}
length_ecoli_genes <- as.numeric(summary(ecoli_cds)[,1])
length_aaceti_genes <- as.numeric(summary(aaceti_cds)[,1])
cds_table_total <- data.table(Organism = c("E.coli", "A.aceti"), "Total Amount of Coding Sequence" = c(sum(length_ecoli_genes), sum(length_aaceti_genes)))
print(cds_table_total)
```
# Describe any differences between the two organisms. 
```{r percentage increase in total amount of coding sequences}
percentage_increase_total <- (sum(length_ecoli_genes)-sum(length_aaceti_genes))/sum(length_aaceti_genes)*100
print(percentage_increase_total)
```
In line with a 24.6% increase in genes, E.coli contains a proportional increase in total coding sequence, with an increase of 22.9% above the total amount of coding DNA contained in A.aceti organisms.


# 3. Calculate the length of all coding sequences in these two organisms. Make a boxplot of coding sequence length in these organisms. What is the mean and median coding sequence length of these two organisms? Describe any differences between the two organisms.
```{r Boxplot of coding sequence length in these organisms}
boxplot(length_ecoli_genes, length_aaceti_genes, names = c("E.coli", "A.aceti"), main = "Coding DNA Sequence Length in E.coli and A.aceti", ylab="Length (bp)")
```

```{r Mean and median coding sequence length in these organisms}
cds_table_stats <- data.table(Organism = c("E.coli", "A.aceti"), "Mean cDNA Sequence Length" = c(mean(length_ecoli_genes), mean(length_aaceti_genes)), "Median cDNA Sequence Length" = c(median(length_ecoli_genes), median(length_aaceti_genes)))
print(cds_table_stats)
```

# Describe any differences between the two organisms. 


```{r}
d_ecoli <- density(length_ecoli_genes)
d_aaceti <- density(length_aaceti_genes)
plot(d_ecoli, col="red", main = "Density plot of coding DNA sequence lengths", xlab = "cDNA length")
lines(d_aaceti, col="blue")
text(x = 6000, y = 6e-04, labels = "E.coli", col = "red", cex = 1.2)
text(x = 6000, y = 5e-04, labels = "A.aceti", col = "blue", cex = 1.2)
```




# 4. Calculate the frequency of DNA bases in the total coding sequences for both organisms. Perform the same calculation for the total protein sequence. Create bar plots for nucleotide and amino acid frequency. Describe any differences between the two organisms.
```{r bar plots for frequency of DNA bases in the total coding sequences for both organisms}

```

```{r bar plots for frequency of amino acids in the total protein sequences for both organisms}

```

# Describe any differences between the two organisms. 

# 5. Create a codon usage table and quantify the codon usage bias among all coding sequences. Describe any differences between the two organisms with respect to their codon usage bias. Provide charts to support your observations.
```{r}

```


# 6. In the organism of interest, identify 10 protein sequence k-mers of length 3-5 which are the most over- and under-represented k-mers in your organism of interest. Are these k-mers also over- and under-represented in E. coli to a similar extent? Provide plots to support your observations. Why do you think these sequences are present at different levels in the genomes of these organisms?
```{r}

```


```{r Recording Session Info}
sessionInfo()
```