---
title: "SLE777_Renee Carroll_Assessment 4"
author: "Renee Carroll"
date: "2025-09-25"
output: 
  html_document:
    toc: yes
    fig_width: 7
    fig_height: 7
theme: yeti
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

## Part 1
The file “gene_expression.tsv” contains RNA-seq count data for three samples of interest.

# 1. Read in the file, making the gene identifiers the row names. Show a table of values for the first six genes.

```{r download files from GitHub, and show first 6 genes}
URL1 = "https://raw.githubusercontent.com/ghazkha/Assessment4/refs/heads/main/gene_expression.tsv"
download.file(URL1, destfile="gene_expression.tsv")
gene_expression <- read.delim("gene_expression.tsv", header = TRUE, row.names = 1, sep = "\t", stringsAsFactors = FALSE)
head(gene_expression)
```

# 2. Make a new column which is the mean of the other columns. Show a table of values for the first six genes.

```{r mean saved as new column}
gene_expression$mean_expression <- rowMeans(gene_expression)
head(gene_expression)
```

# 3. List the 10 genes with the highest mean expression

```{r 10 genes with highest mean - whole table}
ordered_expression <- gene_expression[order(-gene_expression$mean_expression),]
genes_highest_expression <- head(ordered_expression, 10)
genes_highest_expression
```

# Listing just the 10 gene names

```{r 10 genes with highest mean - just names as a list}
rownames(genes_highest_expression)
```

# 4. Determine the number of genes with a mean < 10

```{r number of genes with low expression < 10}
low_exp_genes <- subset(gene_expression, mean_expression<10)
nrow(low_exp_genes)
```

# 5. Make a histogram plot of the mean values and include it into your report.

```{r histogram}
hist(gene_expression$mean_expression, breaks=1000, main = "Mean Gene Expression", xlab="Mean Gene Expression", col="dodgerblue2")
```

As the data is highly positively skewed, it is best represented on a logarithmic scale.


```{r histogram of log of values}
gene_subset_expressionhigherthan10 <- subset(gene_expression, mean_expression>10)
log_gene_subset_expressionhigherthan10 <- log(gene_subset_expressionhigherthan10)
hist(log_gene_subset_expressionhigherthan10$mean_expression, breaks=50, main = "Mean Gene Expression", xlab = "Log Mean Gene Expression", col = "dodgerblue2")
```

Try that again, using log+1, as log of 0 gives infinity... does this make a difference to how the graph looks?


```{r histogram of log of values+1}
gene_subset_expressionhigherthan10 <- subset(gene_expression, mean_expression>10)
log_gene_subset_expressionhigherthan10 <- log(gene_subset_expressionhigherthan10+1)
hist(log_gene_subset_expressionhigherthan10$mean_expression, breaks=50, main = "Mean Gene Expression", xlab = "Log Mean Gene Expression", col = "dodgerblue2")
```

## Part 2
The file “growth_data.csv” contains measurements for tree circumference growing at two sites, control site and treatment site which were planted 20 years ago.

# 6. Import this csv file into an R object. What are the column names?

```{r Import csv file}
URL2 = "https://raw.githubusercontent.com/ghazkha/Assessment4/refs/heads/main/growth_data.csv"
download.file(URL2, destfile="growth_data.csv")
growth_data <- read.csv("growth_data.csv") # read.csv automatically treats the first row as headings, therefore you don't need to specify it in the code
```

```{r determine column names}
colnames(growth_data)
```





# 7. Calculate the mean and standard deviation of tree circumference at the start and end of the study at both sites.

```{r northeast site, mean of tree circumferences, start of study}
mean(subset(growth_data, Site == "northeast")$Circumf_2005_cm)
```

```{r northeast site, standard deviation of tree circumferences, start of study}
sd(subset(growth_data, Site == "northeast")$Circumf_2005_cm)
```

```{r northeast site, mean of tree circumferences, end of study}
mean(subset(growth_data, Site == "northeast")$Circumf_2020_cm)
```

```{r northeast site, standard deviation of tree circumferences, end of study}
sd(subset(growth_data, Site == "northeast")$Circumf_2020_cm)
```

```{r southwest site, mean of tree circumferences, start of study}
mean(subset(growth_data, Site == "southwest")$Circumf_2005_cm)
```

```{r southwest site, standard deviation of tree circumferences, start of study}
sd(subset(growth_data, Site == "southwest")$Circumf_2005_cm)
```

```{r southwest site, mean of tree circumferences, end of study}
mean(subset(growth_data, Site == "southwest")$Circumf_2020_cm)
```

```{r southwest site, standard deviation of tree circumferences, end of study}
sd(subset(growth_data, Site == "southwest")$Circumf_2020_cm)
```

# 8. Make a box plot of tree circumference at the start and end of the study at both sites.

```{r boxplot of tree circumference at the start and end of the study at both sites}
boxplot(subset(growth_data, Site == "northeast")$Circumf_2005_cm, subset(growth_data, Site == "northeast")$Circumf_2020_cm, subset(growth_data, Site == "southwest")$Circumf_2005_cm, subset(growth_data, Site == "southwest")$Circumf_2020_cm, names = c("Northeast 2005", "Northeast 2020", "Southwest 2005", "Southwest 2020"), ylab = "Circumference (cm)", main = "Increase in tree girth between 2005 and 2020, at each study site")
```

# 9. Calculate the mean growth over the last 10 years at each site.

```{r mean growth over the last 10 years, northeast site}
growth_data$growth_2010_2020 <- growth_data$Circumf_2020_cm-growth_data$Circumf_2010_cm
mean(subset(growth_data, Site == "northeast")$growth_2010_2020)
```
```{r mean growth over the last 10 years, southwest site}
mean(subset(growth_data, Site == "southwest")$growth_2010_2020)
```

# 10. Use the t.test to estimate the p-value that the 10 year growth is different at the two sites.
The t.test() function returns an object containing various information, including:
t-statistic: The calculated t-value.
df: Degrees of freedom.
p-value: The probability of observing a t-statistic as extreme as, or more extreme than, the calculated one, assuming the null hypothesis is true. A small p-value (typically < 0.05) suggests evidence against the null hypothesis.
confidence interval: A range within which the true population mean difference (for two-sample tests) or population mean (for one-sample tests) is likely to fall.
sample means: The means of the groups or sample.

Assumptions:
T-tests rely on assumptions like normality of data (or residuals for two-sample tests) and, for independent samples, potentially equality of variances. You may need to check these assumptions before interpreting the results.

```{r test for normality in the northeast data}
hist(subset(growth_data, Site == "northeast")$growth_2010_2020, breaks = 20)
```

```{r test for normality in the southwest data}
hist(subset(growth_data, Site == "southwest")$growth_2010_2020, breaks = 20)
```


```{r t.test, not assuming equal variance (Welch t.test)}
t.test(growth_2010_2020 ~ Site, data = growth_data, var.equal = FALSE) #var.equal = TRUE: Include this if you assume equal variances between the groups (for a Student's t-test). Omit or set to FALSE for Welch's t-test (which doesn't assume equal variances).
```
```{r t.test, assuming equal variance}
t.test(growth_2010_2020 ~ Site, data = growth_data, var.equal = TRUE) #var.equal = TRUE: Include this if you assume equal variances between the groups (for a Student's t-test). Omit or set to FALSE for Welch's t-test (which doesn't assume equal variances).
```






```{r Recording Session Info}
sessionInfo()
```